packages/@blockprotocol/hook/src/react.ts

```diff
@@ -188,26 +188,16 @@ export const useHook = <T extends HTMLElement>(
       return;
     }
 
+    const existingHookId = existingHookRef.current?.id;
+
     const teardownPromise =
       existingHookRef.current?.teardown?.().catch() ?? Promise.resolve();
 
     if (node && service) {
       const controller = new AbortController();
 
-      /**
-       * Is this an update to the existing hook, or is it a whole new hook? The
-       * only param to the hook which can change without creating a new hook is
-       * the node. Any other change will result in a new hook being created
-       */
-      const reuseId =
```

Comment:
This was the original proposal – have agreed with @nathggns that we can just allow any change in param to be an update instead of requiring teardown + recreation